### ISSUE TITLE
bisync: respect --backup-dir

### ISSUE THREAD

**Issue by ivandeex** [rclone/rclone#5685]
## Synopsis

bisync should support the `--backup-dir` flag similar to `sync`.

### https://github.com/rclone/rclone/pull/5164#pullrequestreview-729822158 (ncw)

Will users be able to use something like `--backup-dir` or `--suffix`? I can see users wanting that to prevent data loss. Maybe `--suffix` works already and you could filter the backups out using the filtering? Perhaps this should be automatic.

### https://github.com/rclone/rclone/pull/5164#issuecomment-842407758 (JWink3101)

Personally, I'd **much** rather see `--backup-dir` support out of the gate with that being the check. And again, it matches current rclone behavior.

## How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by acegene** [comment]
Behavior with rclone v1.64.2 seems to silently ignore the flag. Should it be changed to give a warning/error?

Bumping as this feature is especially useful given the beta nature of bisync.

**Comment by nielash** [comment]
@acegene I have great news for you -- this will be fixed by #7410 (specifically https://github.com/rclone/rclone/pull/7410/commits/6109b14aa34c60649c92741c85164a1f2f3fe9ac.) Any chance you'd like to beta test it? 😃 

The underlying issue was the same one as `--track-renames` -- deletes being handled in separate operations (not using sync). Technically, `--backup-dir` did work before if you were overwriting an existing file, but not if deleting (or renaming) a file. After #7410 it will work for everything, as one would expect.

**Comment by acegene** [comment]
> Technically, --backup-dir did work before if you were overwriting an existing file, but not if deleting (or renaming) a file. After https://github.com/rclone/rclone/pull/7410 it will work for everything, as one would expect.

Thanks for clarifying this.

> I have great news for you -- this will be fixed by https://github.com/rclone/rclone/pull/7410 (specifically https://github.com/rclone/rclone/commit/6109b14aa34c60649c92741c85164a1f2f3fe9ac.) Any chance you'd like to beta test it? 😃

@nielash I appreciate the effort and this usecase I am highly interested in. Does beta testing require building from source? Or will the instruction `sudo -v ; curl https://rclone.org/install.sh | sudo bash -s beta` be applicable?

**Comment by nielash** [comment]
Thanks! The `install.sh` script won't work until the PR is merged into master (after that it should work). In the meantime, installing from source is an option, or I could upload precompiled binaries to my fork as a pre-release. That might be easiest.

**Comment by acegene** [comment]
> I could upload precompiled binaries to my fork as a pre-release. That might be easiest.

If you dont mind this would be easier for me! And despite my sharing of the bash install cmd im likely to get more thorough testing done using windows binaries in case thats an option.

**Comment by nielash** [comment]
> If you dont mind this would be easier for me! And despite my sharing of the bash install cmd im likely to get more thorough testing done using windows binaries in case thats an option.

Sure thing! Just uploaded [versions for every platform here](https://github.com/nielash/rclone/releases/tag/v1.65.0-beta.7496.004c12643.bisync-logger-alpha) (that makefile is pretty cool!)

I also just added another [commit](https://github.com/nielash/rclone/commit/004c12643db1b6d25429f2264d77945a87591daa) to allow two `--backup-dir` paths on different remotes (otherwise `--backup-dir` would only work if `Path1` and `Path2` were different paths on the same remote.) Details in the [docs](https://github.com/nielash/rclone/blob/bisync-logger-alpha/docs/content/bisync.md#--backup-dir1-and---backup-dir2).

Hopefully goes without saying, but: this is just an early preview, not yet reviewed or tested by anyone but me. It may still have bugs, and will probably change somewhat before it is released!
